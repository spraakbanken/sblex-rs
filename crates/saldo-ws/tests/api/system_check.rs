use crate::conftest::TestApp;
use serde_json::Value as JsonValue;

#[tokio::test]
async fn health_check_works() -> eyre::Result<()> {
    // Arrange
    let ctx = TestApp::spawn_app().await?;
    let client = reqwest::Client::new();

    // Act
    let response = client
        // Use the returned application address
        .get(&format!("{}/health", &ctx.address))
        .send()
        .await
        .expect("Failed to execute request.");

    // Assert
    assert!(response.status().is_success());
    let data: JsonValue = response.json().await?;
    insta::assert_json_snapshot!(data);
    Ok(())
}

#[tokio::test]
async fn version_works() -> eyre::Result<()> {
    // Arrange
    let ctx = TestApp::spawn_app().await?;
    let client = reqwest::Client::new();

    // Act
    let response = client
        // Use the returned application address
        .get(&format!("{}/version/json", &ctx.address))
        .send()
        .await
        .expect("Failed to execute request.");

    // Assert
    assert!(response.status().is_success());
    let data = response
        .json::<JsonValue>()
        .await
        .expect("Failed to parse json");
    insta::assert_json_snapshot!(data);
    Ok(())
}
