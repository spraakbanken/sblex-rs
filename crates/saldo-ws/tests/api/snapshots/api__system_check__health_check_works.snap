---
source: crates/saldo-ws/tests/api/system_check.rs
expression: data
---
{
  "status": "UP"
}
